python/src/uagents/agent.py

```diff
@@ -1147,7 +1151,10 @@ def __init__(
         self._port = port or 8000
         self._queries: Dict[str, asyncio.Future] = {}
         self._logger = get_logger("bureau", log_level)
-        self._server = ASGIServer(self._port, self._loop, self._queries, self._logger)
+        self._dispenser = Dispenser()
```

Comment:
Have you tested this with the bureau? I don't think this will work properly as each Agent already has a Dispenser which is used for sending messages. We need to trace where these messages will end up and potentially reroute the messages somehow to the bureaus instance.